Library/Homebrew/cmd/update.sh

```diff
@@ -302,6 +312,9 @@ EOS
   # kill all of subprocess on interrupt
   trap '{ pkill -P $$; wait; exit 130; }' SIGINT
 
+  HOMEBREW_UPDATE_FAILED_FILE="$HOMEBREW_REPOSITORY/.git/UPDATE_FAILED"
```

Comment:
Unless that doesn't work with the subshell it is used with later on. Not entirely sure about that …